bigframes/core/rewrite.py

```diff
@@ -126,8 +137,10 @@ def maybe_join(
         new_columns = remap_names(join_def, lselection, rselection)
 
         # Reconstruct ordering
+        reverse_root = self.reverse_root
         if join_type == "right":
             new_ordering = right.ordering
+            reverse_root = right.reverse_root
         elif join_type == "outer":
```

Comment:
do we need to redefine the `reverse_root` for the outer join case?

Reply:
outer join we treat as left order taking precedence, so it inherits reverse_root from the self.